src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -384,15 +385,26 @@ internal partial class UnboundLambda
             ImmutableArray<TypeWithAnnotations> types,
             ImmutableArray<string> names,
             ImmutableArray<bool> discardsOpt,
+            ImmutableArray<EqualsValueClauseSyntax?> defaultValues,
             bool isAsync,
             bool isStatic)
         {
             Debug.Assert(binder != null);
             Debug.Assert(syntax.IsAnonymousFunction());
             bool hasErrors = !types.IsDefault && types.Any(static t => t.Type?.Kind == SymbolKind.ErrorType);
 
+            // If we have a parenthesized lambda or anonymous method expression,
+            // we could have default parameters and thus we include the parameter list syntax
+            // to re-use existing binder code for type-checking default params
+            var paramSyntaxList = syntax switch
+            {
+                ParenthesizedLambdaExpressionSyntax parenLam => parenLam.ParameterList,
+                AnonymousMethodExpressionSyntax anonMethod => anonMethod.ParameterList,
+                _ => null
```

Comment:
We shouldn't use a catch-all. We should list the explicit cases we expect to see, and throw `ExceptionUtilities.UnexpectedValue` for cases we didn't expect to see.